### PR TITLE
fix: point to most recent version of the spec

### PR DESCRIPTION
**Pull request by dbluhm**
I'm not sure if now is the appropriate time for this or not considering the current status of the 2.1 specification but, if it is, I figured this would be useful to have the didcomm.org site point to the 2.1 version of the spec.

### CODE REVIEW

**Review by frostyfrog — APPROVED** [review]
This looks good to me, though I would recommend waiting for the Specification to reach [DIF-Ratified Specification](https://github.com/decentralized-identity/org/blob/master/work-item-lifecycle.md#73-working-group-approval) status first. More-so as a formality than anything else, but also, I feel it's important to wait for the final approval.

**Comment by dbluhm** [comment]
Is that something we expect to happen soon?

**Comment by TelegramSam** [comment]
Discussed WG 20240108.

**Pull request merged**
